src/sentry/sentry_metrics/indexer/base.py

```diff
@@ -51,6 +54,19 @@ def from_string(cls: Type[KR], key: str, id: int) -> KR:
         return cls(int(org_id), string, id)
 
 
+@dataclass(frozen=True)
+class UseCaseKeyResult:
```

Comment:
Why do we still need `KeyResult` if we are moving to a use case aware system ?

Reply:
The rest of the system right now are still using it, it will get remove once we start using `UseCaseKeyResult`